src/slopes.jl

```diff
@@ -0,0 +1,228 @@
+# Reference : Dietmar Ratz - An Optimized Interval Slope Arithmetic and its Application
+using IntervalArithmetic, ForwardDiff, StaticArrays
+import Base: +, -, *, /, ^, sqrt, exp, log, sin, cos, tan, asin, acos, atan
+import IntervalArithmetic: mid, interval
+
+"""
+Expands the slope of `f` over `x` at the point `c` (default `c = mid(x)`)
+"""
+function slope(f::Function, x::Interval, c::Number = mid(x))
+    try
+        f(slope_var(x, c)).s
+    catch y
+        if isa(y, MethodError)
+            ForwardDiff.derivative(f, x)
+        end
+    end
+end
+
+function slope(f::Function, x::Union{IntervalBox, SVector}, c::AbstractVector = mid.(x)) # multidim
```

Comment:
So you don't have a separate type representing a multidimensional slope? Probably you should, and it should probably be an `SVector` or `IntervalBox`?

Reply:
I will have to write overloading rules for all of them again, and most implementations use the result of the 1D `slope` function to construct the matrix for the multidimensional version. So it will be really complicated to have a new type for multidimensional slope. The function returns an `SArray` as is.